ffmpeg-plugin/mcm_tx.c

```diff
@@ -0,0 +1,187 @@
+#include "libavutil/log.h"
+#include "libavutil/opt.h"
+#include "libavutil/internal.h"
+#include "libavutil/pixdesc.h"
+#include "libavformat/avformat.h"
+#include "libavformat/mux.h"
+#include <mcm_dp.h>
+#include <bsd/string.h>
+
+typedef struct McmMuxerContext {
+    const AVClass *class; /**< Class for private options. */
+
+    /* arguments */
+    char *ip_addr;
+    char *port;
+    char *payload_type;
+    char *protocol_type;
+    int width;
+    int height;
+    enum AVPixelFormat pixel_format;
+    AVRational frame_rate;
+    char *socket_name;
+    int interface_id;
+
+    mcm_conn_context *tx_handle;
+} McmMuxerContext;
```

Comment:
The same as above - can we have this in any way more dependent on existing structures and have this moved to separate include file in sdk?

Reply:
Let's do the refactoring in a separate PR. I suggest not to amend this one.